shared/actions/chat.js

```diff
@@ -0,0 +1,303 @@
+// @flow
+import * as Constants from '../constants/chat'
+import engine from '../engine'
+import {CommonMessageType, CommonTLFVisibility, LocalMessageUnboxedState, NotifyChatChatActivityType, localGetInboxAndUnboxLocalRpcPromise, localGetThreadLocalRpcPromise, localPostLocalRpcPromise} from '../constants/types/flow-types-chat'
+import {List, Map} from 'immutable'
+import {call, put, select} from 'redux-saga/effects'
+import {takeLatest, takeEvery} from 'redux-saga'
+
+import type {ConversationIDKey, InboxState, IncomingMessage, LoadInbox, LoadMoreMessages, LoadedInbox, Message, PostMessage, SelectConversation, SetupNewChatHandler} from '../constants/chat'
+import type {GetInboxAndUnboxLocalRes, IncomingMessage as IncomingMessageRPCType, MessageUnboxed} from '../constants/types/flow-types-chat'
+import type {SagaGenerator} from '../constants/types/saga'
+import type {TypedState} from '../constants/reducer'
+
+const {conversationIDToKey, keyToConversationID, InboxStateRecord} = Constants
+
+function postMessage (conversationIDKey: ConversationIDKey, text: string): PostMessage {
+  return {type: Constants.postMessage, payload: {conversationIDKey, text}}
+}
+
+function setupNewChatHandler (): SetupNewChatHandler {
+  return {type: Constants.setupNewChatHandler, payload: undefined}
+}
+
+function loadInbox (): LoadInbox {
+  return {type: Constants.loadInbox, payload: undefined}
+}
+
+function loadMoreMessages (): LoadMoreMessages {
+  return {type: Constants.loadMoreMessages, payload: undefined}
+}
+
+function selectConversation (conversationIDKey: ConversationIDKey): SelectConversation {
+  return {type: Constants.selectConversation, payload: {conversationIDKey}}
+}
+
+// This is emoji aware hence all the weird ... stuff. See https://mathiasbynens.be/notes/javascript-unicode#iterating-over-symbols
+function _makeSnippet (message, max) {
+  // $FlowIssue flow doesn't understand spread + strings
+  return [...(message.substring(0, max * 4).replace(/\s+/g, ' '))].slice(0, max).join('')
+}
+
+function _inboxToConversations (inbox: GetInboxAndUnboxLocalRes): List<InboxState> {
+  return List((inbox.conversations || []).map(convo => {
+    const recentMessage: ?MessageUnboxed = (convo.maxMessages || []).find(message => (
+      message.state === LocalMessageUnboxedState.valid &&
+      message.valid &&
+      message.valid.messageBody.messageType === CommonMessageType.text
```

Comment:
i just tried to do this but that means i need to map the entire thing or do more logic to find it (keeping this logic) then just converting it to just pull out the string. i think this is very specific to how the inbox works and likely won't occur elsewhere)